src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java

```diff
@@ -263,9 +259,36 @@ private BuildConfigProjectProperty buildConfigProjectForJob(WorkflowJob job) {
 
   private void upsertBuildConfigForJob(WorkflowJob job, BuildConfigProjectProperty buildConfigProjectProperty) {
     boolean create = false;
+
+    logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+      buildConfigProjectProperty.getName());
+
     BuildConfig jobBuildConfig = getOpenShiftClient().buildConfigs().
             inNamespace(buildConfigProjectProperty.getNamespace()).withName(buildConfigProjectProperty.getName()).get();
+
+    if (jobBuildConfig == null){
+
+      logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+        buildConfigProjectProperty.getName());
+
+      logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
+        " with label gitRepository: " + buildConfigProjectProperty.getName());
+
+      BuildConfigList jobBuildConfigs = getOpenShiftClient().buildConfigs().
+        inNamespace(buildConfigProjectProperty.getNamespace())
+        .withLabels(Collections.singletonMap(OPENSHIFT_LABELS_BUILD_CONFIG_GIT_REPOSITORY_NAME, buildConfigProjectProperty.getName())).list();
+
+      if (jobBuildConfigs.getItems().size() == 1){
```

Comment:
So duplicate labels are certainly possible generically speaking ... does the tooling ensure that does not happen?

If the tooling does not ensure this, my opinion, at least getting 1 from a list would be better than the duplicate job creation, no?

If you all agree, let's change the conditional to greater than 0

Reply:
So the logic is you can't create two repositories with the same name, and launcher creates a pipeline for one repository and so it will be always one.

If you say more than one pipeline is there then how you will map?

Reply:
Your first sentence may have answered my question, but I'll rephrase to make sure  ... will the launcher *INSURE* that it only created 1 pipeline for 1 repoistory?

I don't know enough about the launcher to know if it handles 2 users simultaneously trying to create a pipeline off of the same repo.

If the above statements are *NOT* true ...

I think we would rather simply always map the first one, i.e. arbitrarily pick one, but log that more than one exists, and the launcher should be investigated, vs. create 2 jobs if the list is greater than one.

See what I'm saying ?

Reply:
Launcher always creates one build config for one repository created there. Also if two users are simultaneously then they will be in a different namespace and also will be provided different Jenkins instance.

Reply:
ok I think I'm satisfied then thanks  ... maybe a warning log if the size is greater than zero, but I'm fine if you do not want to do that 

please do capture that explanation in a code comment as well

what we'll have to keep an eye on is whether some path outside of fabric's launcher arises in the future and exposes this path